extensions/io/org.eclipse.smarthome.io.mqttembeddedbroker/src/main/java/org/eclipse/smarthome/io/mqttembeddedbroker/internal/EmbeddedBrokerService.java

```diff
@@ -0,0 +1,287 @@
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.io.mqttembeddedbroker.internal.MqttEmbeddedBrokerDetectStart.MqttEmbeddedBrokerStartedListener;
+import org.eclipse.smarthome.io.mqttembeddedbroker.internal.MqttEmbeddedBrokerMetrics.BrokerMetricsListener;
+import org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection;
+import org.eclipse.smarthome.io.transport.mqtt.MqttConnectionObserver;
+import org.eclipse.smarthome.io.transport.mqtt.MqttConnectionState;
+import org.eclipse.smarthome.io.transport.mqtt.MqttException;
+import org.eclipse.smarthome.io.transport.mqtt.MqttService;
+import org.eclipse.smarthome.io.transport.mqtt.MqttServiceObserver;
+import org.osgi.framework.Constants;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.moquette.BrokerConstants;
+import io.moquette.server.Server;
+import io.moquette.server.config.MemoryConfig;
+import io.moquette.spi.security.IAuthorizator;
+
+/**
+ * The {@link EmbeddedBrokerService} starts the embedded broker, creates a
+ * {@link MqttBrokerConnection} and adds it to the {@link MqttService}.
+ *
+ * TODO: wait for NetworkServerTls implementation to enable secure connections as well
+ *
+ * @author David Graeff - Initial contribution
+ */
+@Component(immediate = true, service = EmbeddedBrokerService.class, configurationPid = {
+        "org.eclipse.smarthome.mqttembeddedbroker" }, property = {
+                Constants.SERVICE_PID + "=org.eclipse.smarthome.mqttembeddedbroker",
+                ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI + "=mqtt:mqttembeddedbroker",
+                ConfigurableService.SERVICE_PROPERTY_CATEGORY + "=MQTT",
+                ConfigurableService.SERVICE_PROPERTY_LABEL + "=MQTT Embedded Broker" })
+public class EmbeddedBrokerService implements ConfigurableService, MqttConnectionObserver, MqttServiceObserver,
+        MqttEmbeddedBrokerStartedListener, BrokerMetricsListener {
+    private MqttService service;
+    // private NetworkServerTls networkServerTls; //TODO wait for NetworkServerTls implementation
+
+    private static final String CLIENTID = "embedded-mqtt-broker";
+    protected Server server;
+    private final Logger logger = LoggerFactory.getLogger(EmbeddedBrokerService.class);
+    protected MqttEmbeddedBrokerDetectStart detectStart = new MqttEmbeddedBrokerDetectStart(this);
+    protected MqttEmbeddedBrokerMetrics metrics = new MqttEmbeddedBrokerMetrics(this);
+
+    public MqttBrokerConnection connection;
+
+    @Reference(cardinality = ReferenceCardinality.MANDATORY)
+    public void setMqttService(MqttService service) {
```

Comment:
please also add the unbind method

Reply:
I thought, an unbind method is not required if `ReferenceCardinality.MANDATORY` is set, because the entire service is teared down if the mandatory referenced service died.